### PR TITLE
Prune deprecated EarlyStopping(mode='auto')

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Removes `EarlyStopping(mode='auto')`

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/callbacks/early_stopping.py [hunk 1]
```diff
@@ -40,23 +40,18 @@ class EarlyStopping(Callback):
         patience: number of validation epochs with no improvement
             after which training will be stopped. Default: ``3``.
         verbose: verbosity mode. Default: ``False``.
-        mode: one of {auto, min, max}. In `min` mode,
+        mode: one of {min, max}. In `min` mode,
```
Comment (s-rog) [inline pytorch_lightning/callbacks/early_stopping.py:43]:
```suggestion
        mode: one of ``'min'``, ``'max'``}. In `min` mode,
```

File: pytorch_lightning/callbacks/early_stopping.py [hunk 2]
```diff
@@ -40,23 +40,18 @@ class EarlyStopping(Callback):
...
             training will stop when the quantity
             monitored has stopped decreasing; in `max`
             mode it will stop when the quantity
-            monitored has stopped increasing; in `auto`
-            mode, the direction is automatically inferred
-            from the name of the monitored quantity.
-
-            .. warning::
-               Setting ``mode='auto'`` has been deprecated in v1.1 and will be removed in v1.3.
+            monitored has stopped increasing.
```
Comment (s-rog) [inline pytorch_lightning/callbacks/early_stopping.py:47]:
```suggestion
            monitored has stopped increasing. Default: ``'min'``
```

**Review by s-rog — APPROVED** [review]

File: pytorch_lightning/callbacks/early_stopping.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/callbacks/early_stopping.py:47]: Actually, I'd rather remove the defaults as they appear already in the docs from the function signature.

But in another PR. Feel free to take it.

File: pytorch_lightning/callbacks/early_stopping.py [hunk 3]
```diff
@@ -40,23 +40,18 @@ class EarlyStopping(Callback):
         patience: number of validation epochs with no improvement
             after which training will be stopped. Default: ``3``.
         verbose: verbosity mode. Default: ``False``.
-        mode: one of {auto, min, max}. In `min` mode,
+        mode: one of {min, max}. In `min` mode,
             training will stop when the quantity
             monitored has stopped decreasing; in `max`
```
Comment (carmocca) [inline pytorch_lightning/callbacks/early_stopping.py:45]:
```suggestion
            monitored has stopped decreasing; in ``max``
```

File: pytorch_lightning/callbacks/early_stopping.py [hunk 4]
```diff
@@ -40,23 +40,18 @@ class EarlyStopping(Callback):
         patience: number of validation epochs with no improvement
             after which training will be stopped. Default: ``3``.
         verbose: verbosity mode. Default: ``False``.
-        mode: one of {auto, min, max}. In `min` mode,
+        mode: one of ``'min'``, ``'max'``. In ``min`` mode,
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/early_stopping.py:43]:
```suggestion
        mode: one of ``'min'``, ``'max'``. In ``'min'`` mode,
```

File: pytorch_lightning/callbacks/early_stopping.py [hunk 5]
```diff
@@ -40,23 +40,18 @@ class EarlyStopping(Callback):
...
             training will stop when the quantity
-            monitored has stopped decreasing; in `max`
+            monitored has stopped decreasing; in ``max``
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/early_stopping.py:45]:
```suggestion
            monitored has stopped decreasing and in ``'max'``
```

File: pytorch_lightning/callbacks/early_stopping.py [hunk 6]
```diff
@@ -40,23 +40,18 @@ class EarlyStopping(Callback):
...
-               Setting ``mode='auto'`` has been deprecated in v1.1 and will be removed in v1.3.
+            monitored has stopped increasing.
 
         strict: whether to crash the training if `monitor` is
             not found in the validation metrics. Default: ``True``.
 
     Raises:
         MisconfigurationException:
-            If ``mode`` is none of ``"min"``, ``"max"``, and ``"auto"``.
+            If ``mode`` is none of ``"min"``, ``"max"``.
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/early_stopping.py:54]:
```suggestion
            If ``mode`` is none of ``"min"`` or ``"max"``.
```

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Pull request merged**